lib/src/list.dart

```diff
@@ -30,7 +30,7 @@ import 'results.dart';
 /// added to or deleted from the collection or from the Realm.
 ///
 /// {@category Realm}
-abstract class RealmList<T extends Object> with RealmEntity implements List<T>, Finalizable {
+abstract class RealmList<T extends Object?> with RealmEntity implements List<T>, Finalizable {
```

Comment:
This liberalizes the type args on `RealmList` to support nullable objects. It doesn't actually add support for lists of nullables - that is done in https://github.com/realm/realm-dart/pull/708, but is necessary to simplify a lot of the generic constraints elsewhere.